packages/back-end/src/services/features.ts

```diff
@@ -44,6 +50,57 @@ function generatePayload(
   return defs;
 }
 
+// TODO Put this in a better place?
+function generateVisualExperimentsPayload(
+  visualExperiments: Array<{
+    experiment: ExperimentInterface;
+    visualChangeset: VisualChangesetInterface;
+  }>,
+  _environment: string,
+  groupMap: GroupMap
+): SDKExperiment[] {
+  return visualExperiments.map(({ experiment: e, visualChangeset: v }) => {
+    const phase: ExperimentPhase | null = e.phases.slice(-1)?.[0] ?? null;
```

Comment:
I know you added optional chaining below, but I think if phase is not defined, we don't want to include the experiment in the payload at all.

Reply:
Updated